appium-tests/common/common.spec.js

```diff
@@ -242,7 +248,6 @@ describe('Contacts Android', function () {
 
     describe('Picking contacts', function () {
         afterEach(function (done) {
-            checkSession(done);
```

Comment:
I don't think this function provides much benefit: it only checks if the first, "self-check"-like test finished properly. If the appium session fails to initiate, the other tests will most certainly fail anyways. In my opinion, it is added complexity.

Let me pose the reverse question to you: why have this check?